docs/generators.md

```diff
@@ -0,0 +1,53 @@
+# Generators
+
+**Generators** are classes that are used to render models for a given language.
+
+## Supported languages
+
+- [JavaScript](../src/generators/javascript/JavaScriptGenerator.ts),
+- [TypeScript](../src/generators/typescript/TypeScriptGenerator.ts),
+- [Java](../src/generators/java/JavaGenerator.ts).
+
+## Generator's options
+
+Each generator extends default options for the generator. It means that the generator can also have additional options.
+
+Options is passed as first argument to generator's constructor. Check the example:
+
+```ts
+const generator = new TypeScriptGenerator({ ...options });
+```
+
+Default options contains:
+
+| Option | Type | Description | Default value |
+|---|---|---|---|
+| `indentation` | Object | Options for indentation. | - |
+| `indentation.type` | String | Type of indentation. Its value can be either `SPACES` or `TABS`. | `SPACES` |
+| `indentation.size` | String | Size of indentation. | 2 |
+| `defaultPreset` | Object | Default preset for generator. For more information, read [customisation](./customisation.md) document. | _Implemented by generator_ |
+| `presets` | Array | Array contains **presets**. For more information, read [customisation](./customisation.md) document. | `[]` |
+
+Below is a list with additional options available for a given generator.
+
+### [TypeScript](../src/generators/typescript/TypeScriptGenerator.ts)
+
+| Option | Type | Description | Default value |
+|---|---|---|---|
+| `renderTypes` | Boolean | Renders or not signature for types. | `true` |
+| `modelType` | String | It indicates which model type should be rendered for `object` type. Its value can be either `interface` or `class`. | `class` |
+
+## Custom generator
+
+Creating a new generator which will become a part of the officially supported languages in AsyncAPI Model SDK is pretty simple. The minimum set of required actions to create a new generator are:
```

Comment:
```suggestion
Creating a new generator that will become a part of the officially supported languages in AsyncAPI Model SDK is pretty simple. The minimum set of required actions to create a new generator are:
```

`is pretty simple` do we need this 😄 pretty subjective 😄

Reply:
I left only second sentence here and added at the end of file this

> If you created a generator then you can contribute it to the AsyncAPI Model SDK and it will become the official supported generator.

It is ok for you?